java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -2167,140 +2279,179 @@ public void testVectorLoadUnload() {
+    }
 
-      setVector(vector, STR1, STR2, STR3, STR4, STR5, STR6);
-      vector.setValueCount(15);
+    try (final ViewVarCharVector vector = new ViewVarCharVector("myvector", allocator)) {
+      testFillEmptiesUsageHelper(vector);
+    }
+  }
 
-      /* check the vector output */
-      assertArrayEquals(STR1, vector.get(0));
-      assertArrayEquals(STR2, vector.get(1));
-      assertArrayEquals(STR3, vector.get(2));
-      assertArrayEquals(STR4, vector.get(3));
-      assertArrayEquals(STR5, vector.get(4));
-      assertArrayEquals(STR6, vector.get(5));
+  private void testGetBufferAddress1VarCharVector(VarCharVector vector) {
+    setVector(vector, STR1, STR2, STR3, STR4, STR5, STR6);
+    vector.setValueCount(15);
 
-      List<ArrowBuf> buffers = vector.getFieldBuffers();
-      long bitAddress = vector.getValidityBufferAddress();
-      long offsetAddress = vector.getOffsetBufferAddress();
-      long dataAddress = vector.getDataBufferAddress();
+    /* check the vector output */
+    assertArrayEquals(STR1, vector.get(0));
+    assertArrayEquals(STR2, vector.get(1));
+    assertArrayEquals(STR3, vector.get(2));
+    assertArrayEquals(STR4, vector.get(3));
+    assertArrayEquals(STR5, vector.get(4));
+    assertArrayEquals(STR6, vector.get(5));
 
-      assertEquals(3, buffers.size());
-      assertEquals(bitAddress, buffers.get(0).memoryAddress());
-      assertEquals(offsetAddress, buffers.get(1).memoryAddress());
-      assertEquals(dataAddress, buffers.get(2).memoryAddress());
+    List<ArrowBuf> buffers = vector.getFieldBuffers();
+    long bitAddress = vector.getValidityBufferAddress();
+    long offsetAddress = vector.getOffsetBufferAddress();
+    long dataAddress = vector.getDataBufferAddress();
+
+    assertEquals(bitAddress, buffers.get(0).memoryAddress());
+    assertEquals(offsetAddress, buffers.get(1).memoryAddress());
+    assertEquals(dataAddress, buffers.get(2).memoryAddress());
+    assertEquals(3, buffers.size());
+  }
+
+  private void testGetBufferAddress1ViewVarCharVector(ViewVarCharVector vector) {
+    setVector(vector, STR1, STR2, STR3, STR4, STR5, STR6);
+    vector.setValueCount(15);
+
+    /* check the vector output */
+    assertArrayEquals(STR1, vector.get(0));
+    assertArrayEquals(STR2, vector.get(1));
+    assertArrayEquals(STR3, vector.get(2));
+    assertArrayEquals(STR4, vector.get(3));
+    assertArrayEquals(STR5, vector.get(4));
+    assertArrayEquals(STR6, vector.get(5));
+
+    List<ArrowBuf> buffers = vector.getFieldBuffers();
+    long bitAddress = vector.getValidityBufferAddress();
+    long dataAddress = vector.getDataBufferAddress();
+
+    // ViewVarCharVector only have two buffers
+    // ViewVarCharVector does not have an offset buffer
+    assertEquals(2, buffers.size());
+    assertEquals(bitAddress, buffers.get(0).memoryAddress());
+    assertEquals(dataAddress, buffers.get(1).memoryAddress());
+  }
+
+  @Test /* VarCharVector */
+  public void testGetBufferAddress1() {
```

Comment:
here I did split for readability.

Reply:
One for `ViewVarCharVector` and the other for `VarCharVector`

Reply:
IMO it is less readable that I have to read two very similar blocks of code and jump around. A single use method is not necessarily clearer than just having a long method unless you have some opportunity to document things or something.


If you are worried about the vector-specific code I would do

```java
    try (final VarCharVector vector = new VarCharVector("myvector", allocator)) {
      commonTest(vector);
      assertEquals(/*...type specific tests go here...*/);
    }

    try (final ViewVarCharVector vector = new ViewVarCharVector("myviewvector", allocator)) {
      commonTest(vector);
    }
```

Reply:
Addressed as advised.